app/src/main/java/com/github/bjolidon/bootcamp/ui/authentication/AuthenticationActivity.kt

```diff
@@ -0,0 +1,120 @@
+package com.github.bjolidon.bootcamp.ui.authentication
+
+import android.content.Context
+import android.content.Intent
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.firebase.ui.auth.AuthUI
+import com.firebase.ui.auth.AuthUI.IdpConfig
+import com.firebase.ui.auth.FirebaseAuthUIActivityResultContract
+import com.firebase.ui.auth.data.model.FirebaseAuthUIAuthenticationResult
+import com.github.bjolidon.bootcamp.R
+import com.google.firebase.auth.FirebaseAuth
+
+/**
+ * This activity is used to sign in or sign out the user.
+ * Called by creating an Intent with the extra "signIn" set to true if the user wants to sign in
+ * or false if the user wants to sign out.
+ */
+class AuthenticationActivity: AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_authentication)
+        val signIn = intent.getBooleanExtra("signIn", false)
+        if (signIn) {
+            createSignInIntent()
+        } else {
+            signOut()
+        }
+    }
+
+    private val signInLauncher = registerForActivityResult(
+        FirebaseAuthUIActivityResultContract()
+    ) { res ->
+        this.onSignInResult(res)
+    }
+
+    /**
+     * Create an intent to sign in the user and wait for the result.
+     */
+    private fun createSignInIntent() {
+        // Choose authentication providers
+        val providers = arrayListOf(
+            IdpConfig.GoogleBuilder().build()
+        )
+        // Check if the device is connected to the internet

```

Comment:
This looks like a race condition: the device can lose connection immediately after the check. Since you have to handle the lost connection case anyway, is it useful to do the check now?